### PR TITLE
Parallelize Builds

### PR DESCRIPTION
**Pull request by dawsontoth**
Build for the various targets and architectures in parallel, and then copy over the various files in parallel. Builds take about 30 minutes for me using this, and they _appear_ to work as normal. (I say _appear_
 because I'm just poking around with an app to test.)

### CODE REVIEW

**Comment by dawsontoth** [comment]
Still testing to measure how long building them in series takes on my machine. I know it takes a while... but haven't quantified the speedup yet.

**Comment by dawsontoth** [comment]
Also added a --quiet flag if you don't care about the output from cmake or msbuild.

**Comment by dawsontoth** [comment]
Yeah, this reduced my build time from 50 minutes to 30 minutes.

**Comment by sgtcoolguy** [comment]
@dawsontoth I had previously used the /m flag for msbuild to parallelize the actual compilation and ran into exactly what happened here - it saturated the build agent's CPU causing the java process that communicates with Jenkins master to get starved and the agent to be marked offline mid-build.

That's why the PR build failed...

**Comment by dawsontoth** [comment]
aw :(

**Comment by dawsontoth** [comment]
hold on a sec...

**Comment by dawsontoth** [comment]
There, how about that Chris? So the build server can do its own thing, and when building locally, we can use all the processors.

File: Tools/Scripts/build/build.js [hunk 1]
```diff
@@ -34,71 +162,42 @@ function runCMake(sourceDir, buildDir, buildType, platform, arch, callback) {
...
-	});
-
-	prc.on('close', function (code) {
-		if (code != 0) {
-			callback("Failed to run MSBuild");
-		} else {
-			callback();
-		}
-	});
+	var args = ['/p:Configuration=' + buildType];
```
Comment (garymathews) [inline Tools/Scripts/build/build.js]: Could add ```/m``` here? https://msdn.microsoft.com/en-us/library/bb651793.aspx

**Comment by sgtcoolguy** [comment]
I think the jenkins build agent is hosed right now. We'll need to reboot it before builds will work again, I'm betting

**Comment by dawsontoth** [comment]
sorry if i did that :)

**Comment by sgtcoolguy** [comment]
@dawsontoth Any reason you killed this PR? I'm ok with the idea to help speed up local builds. The last build failed due to an issue copying all the HAL headers over, much like the Windows doc PR. I'm not sure why that's suddenly happening...

**Comment by dawsontoth** [comment]
Pulled in the fix for CI, hopefully it'll pass now.

**Comment by dawsontoth** [comment]
test this please

**Comment by sgtcoolguy** [comment]
test this please.

**Comment by dawsontoth** [comment]
will have to rebase again

**Pull request merged**